app/converters/sync_actual_events_converter.rb

```diff
@@ -0,0 +1,56 @@
+module SyncActualEventsConverter
+  include BodConverter
+
+  def process_sync_noun(noun)
+    begin
+      floc = FunctionalLocation.find_by uuid: noun.dig("actualEvent", "entity", "uUID")
+      unless floc.present?
+        floc = create_missing_floc(noun.dig("actualEvent", "entity"), "SyncActualEvents")
+      end
+
+      noun.dig("actualEvent", "attributeSetForEntity", "attributeSet", "group", "group").each do |group|
+        # Loops through all the attributes in all the groups and adds the actual events to the db.
+        group[:setAttribute].each do |event_info|
+          values = extract_from_value_content(event_info)
+          ActualEvent.create(uuid: noun.dig("actualEvent", "uUID"), group_uuid: group[:uUID], attribute_type: event_info.dig("type","uUID"),
+            value: values[:value], uom: values[:uom], functional_location: floc)
+        end
+
+      end
+
+      # Outer Group => Event Type
+      value = noun.dig("actualEvent","attributeSetForEntity","attributeSet","group","setAttribute","valueContent","enumerationItem","uUID")
+      uom = ''
+      ActualEvent.create(uuid: noun.dig("actualEvent","uUID"), group_uuid: noun.dig("actualEvent","attributeSetForEntity","attributeSet","group","uUID"),
+      attribute_type: attribute.dig("type","uUID"), value: value, uom: uom, functional_location: floc)
+
+      Rails.logger.debug "\n Actual Events successfully added"
+
+    rescue StandardError => e
+      Rails.logger.error("\n Actual Events COULD NOT be added - #{e}")
+    end
+  end
+
+  def extract_from_value_content(event_info)
+    uom = ''
+    case event_info.dig("type","uUID")
```

Comment:
I am working on creating a reference data that refers to CCOM and project specific data.
Will refer to Types from the CCOM data here and not the UUID s directly.